core/resharding.go

```diff
@@ -0,0 +1,173 @@
+package core
+
+import (
+	"math/rand"
+	"sort"
+	"strconv"
+
+	"github.com/harmony-one/harmony/core/types"
+)
+
+const (
+	// InitialSeed is the initial random seed, a magic number to answer everything, remove later
+	InitialSeed int64 = 42
+)
+
+// ShardingState is data structure hold the sharding state
+type ShardingState struct {
+	epoch      uint64 // current epoch
+	rnd        int64  // random seed for resharding
+	numShards  int
+	shardState types.ShardState
+}
+
+// sortedCommitteeBySize will sort shards by size
+// Suppose there are N shards, the first N/2 larger shards are called active committees
+// the rest N/2 smaller committees are called inactive committees
+// actually they are all just normal shards
+func (ss *ShardingState) sortCommitteeBySize() {
+	sort.Slice(ss.shardState, func(i, j int) bool {
+		return len(ss.shardState[i].NodeList) > len(ss.shardState[j].NodeList)
+	})
+}
+
+func (ss *ShardingState) assignNewNodes(newNodeList []types.NodeID) {
```

Comment:
please add some comments to elaborate on this algorithm.

Reply:
I added the comments for this function.